lib/wibox/layout/overflow.lua

```diff
@@ -0,0 +1,549 @@
+            if i > 1 and spacing_widget then
+                table.insert(result, base.place_widget_at(
+                    spacing_widget,
+                    -- The way how spacing is added for regular widgets
+                    -- and the `spacing_widget` is disconnected:
+                    -- The offset for regular widgets is added to `pos` one
+                    -- iteration _before_ the one where the widget is actually
+                    -- placed.
+                    -- Because of that, the placement for the spacing widget
+                    -- needs to substract that offset to be placed right after
+                    -- the previous regular widget.
+                    math.floor(is_y and content_x or (content_x - spacing)),
+                    math.floor(is_y and (content_y - spacing) or content_y),
+                    math.floor(is_y and content_w or spacing),
+                    math.floor(is_y and spacing or content_h)
+                ))
+            end
+
+            table.insert(result, base.place_widget_at(
+                w,
+                math.floor(content_x),
+                math.floor(content_y),
+                math.floor(content_w),
+                math.floor(content_h)
+            ))
+        end
+    end
+
+    return result
+end
+
+function overflow:before_draw_children(_, cr, width, height)
+    -- Clip drawing for children to the space we're allowed to draw in
+    cr:rectangle(0, 0, width, height)
+    cr:clip()
+end
+
+
+--- The amount of units to advance per scroll event.
+--
+-- This affects calls to `scroll` and the default mouse wheel handler.
+--
+-- The default is `10`.
+--
+-- @property step
+-- @tparam number step The step size.
+function overflow:set_step(step)
+    self._private.step = step
+    -- We don't need to emit enything here, since changing step only really
+    -- takes effect the next time the user scrolls
+end
+
+
+--- Scroll the layout's content by `amount * step`.
+--
+-- A positive amount scroll down/right, a negative amount scrolls up/left.
+--
+-- The amount of units scrolled is affected by `step`.
+--
+-- @method overflow:scroll
+-- @tparam number amount The amount to scroll by.
+-- @emits property::overflow::scroll_factor
+-- @emitstparam property::overflow::scroll_factor number scroll_factor The new
+--   scroll factor.
+-- @emits widget::layout_changed
+-- @emits widget::redraw_needed
+function overflow:scroll(amount)
+    if amount == 0 then
+        return
+    end
+    local interval = self._private.used_in_dir
+    local delta = self._private.step / interval
+
+    local factor = self._private.scroll_factor + (delta * amount)
+    self:set_scroll_factor(factor)
+end
+
+
+--- The scroll factor.
+--
+-- The scroll factor represents the how for the layout's content is currently
+-- scrolled. It is represented as a fraction from `0` to `1`, where `0` is the
+-- start of the content and `0` is the end.
```

Comment:
I'm not sure to understand the description here. Can you make it simpler? Something like "The scroll factor represents the layout's current scrolled position.".

Also, I think it's "where `0` is the start of the content and **`1`** is the end."